src/test/scala/com/fluidcode/processing/silver/GetPostsInfoAndCreateTableSpec.scala

```diff
@@ -9,25 +12,83 @@ import org.apache.spark.sql.test.SharedSparkSession
 
 class GetPostsInfoAndCreateTableSpec extends QueryTest
   with SharedSparkSession
-  with DeltaExtendedSparkSession  {
+  with DeltaExtendedSparkSession {
 
   test("getPostsInfo should create post info table from Bronze Layer" ) {
     withTempDir { dir =>
       val sparkSession = spark
       val conf = Configuration(dir.toString)
       conf.init(sparkSession)
 
-      val path = "c:/tmp/StreamSource/silver"
+      import sparkSession.implicits._
+
+      val path = s"${conf.rootPath}/${conf.database}/${conf.sample}"
+
+      val sampleDf = Seq(
+        Data(Array(GraphImagesElements(
+          "Graph1", CommentsData(Array(DataElements(
+            1623779105, "1382894360", OwnerData("138289436000", "https://profile_pic_url1", "mehrez"), "comment_text1")
+          )),
+          comments_disabled = true, DimensionsData(1080, 1920), "https://instagram.url1", Likes(100),
+          Captions(Array(EdgesElements(NodeData("caption_text1")))),
+          Comments(50), null, "image_id1", is_video = false, null, "s564gsd", Owner("138289436"), "shortcode1", Array("tag1"),
+          1623779104, Array(ThumbnailElements(150, 150, "https://instagramthumbnail_src1")), "https://thumbnail_src1",
+          Array("url1", "url2"), "benz")),
+          ProfileInfo(1623779107, Info("biography1", 1000, 500, "full_name1", "138289cc", is_business_account = true, is_joined_recently = false, is_private = false, 100,
+            "https://profile_pic_url1"), "benz"))
+      ).toDF()
+
+      sampleDf.write.format("json").save(path)
+
       val bronzeLayer = new BronzeLayer(conf, sparkSession, path)
       bronzeLayer.createBronzeTable()
 
-      val bronzeData = spark.read.format("delta").load(s"${conf.rootPath}/${conf.database}/${conf.bronzeTable}")
-      val postsInfo = getPostsInfo(bronzeData)
-      createPostsInfoTable(postsInfo, conf)
+      createPostsInfoTable(s"${conf.rootPath}/${conf.database}/${conf.bronzeTable}", conf, spark)
```

Comment:
this function can take "conf" and "spark" only, no need to give it a path as an argument as you are able to extract the path from Configuration.